runtime/src/append_vec.rs

```diff
@@ -230,43 +261,29 @@ impl<'a> StoredAccountMeta<'a> {
     }
 
     fn sanitize(&self) -> bool {
-        self.sanitize_executable()
+        self.sanitize_executable_and_application_fees()
             && self.sanitize_lamports()
-            && self.sanitize_has_application_fees()
-    }
-
-    fn sanitize_executable(&self) -> bool {
-        // Sanitize executable to ensure higher 7-bits are cleared correctly.
-        self.ref_executable_byte() & !1 == 0
+            && self.sanitize_executable_and_application_fees()
     }
 
-    fn sanitize_has_application_fees(&self) -> bool {
+    fn sanitize_executable_and_application_fees(&self) -> bool {
         // Sanitize executable to ensure higher 7-bits are cleared correctly.
-        self.ref_has_application_fees_byte() & !1 == 0
+        self.ref_account_flags_byte() & !3 == 0
```

Comment:
Need to fix the comment since it's now out of date.

If we really want to be explicit of where this value comes from can do something like:
```
    const SANITIZER: u8 = !(AccountFlags::EXECUTABLE.bits() | AccountFlags::HAS_APPLICATION_FEES.bits());
    account_flags_byte & SANITIZER == 0
```